### PR TITLE
Usage Instructions for SAMLAssertion

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
## What Has Changed?

Add a paragraph about how to use the `SAMLAssertion` authentication type.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -77,6 +77,26 @@ DestinationAccessor.getLoader().tryGetDestination(destinationName, options);
 
 See the section on [destination options](#building-destination-options) below.
 
+### Using the SAMLAssertion Authentication Type
+
+Using the [`SAMLAssertion`](https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/d81e1683bd434823abf3ceefc4ff157f.html) authentication types requires a different caching strategy for `HttpClient`s.
+This is, because upon authentication at the target system, a session cookie is returned, which is stored within the used `HttpClient` instance.
```
Comment (KavithaSiva) [inline docs/java/features/connectivity/destinations.mdx:83]:
```suggestion
This is because upon authentication at the target system, a session cookie is returned, which is stored within the used `HttpClient` instance.
```

File: docs/java/features/connectivity/destinations.mdx [hunk 2]
```diff
@@ -77,6 +77,26 @@ DestinationAccessor.getLoader().tryGetDestination(destinationName, options);
...
+For subsequent calls, the service then expects authenticated clients to include their session cookie.
+To not loose the session in between requests, the SAP Cloud SDK offers the `TimeScopedHttpClientCache` (as opposed to the `RequestScopedHttpClientCache`, which is used by default).
```
Comment (KavithaSiva) [inline docs/java/features/connectivity/destinations.mdx:85]: I would rephrase this saying that while using `RequestScopedHttpClientCache`, a new `HttpClient` is created for each request to the application and using `TimeScopedHttpClientCache` ensures that `HttpClient` is retained over a time period, hence ensuring that the session cookies are retained.
I would skip saying that the using `TimeScopedHttpClientCache` helps in retaining the session.

**Review by KavithaSiva — COMMENTED** [review]
Thanks for writing a very good concise section, just few minor comments from my side.

File: docs/java/features/connectivity/destinations.mdx (hunk 2 above)
Comment (KavithaSiva) [inline docs/java/features/connectivity/destinations.mdx:85]: I would skip saying that the using `TimeScopedHttpClientCache` helps in retaining the session, it rather caches the HttpClient for the time interval specified.

Comment (Johannes-Schneider) [reply docs/java/features/connectivity/destinations.mdx]: I rephrased the explanation a bit. @KavithaSiva, please review again.

**Review by KavithaSiva — APPROVED** [review]
LGTM.
